C/BFS_AdjacencyList.c

```diff
@@ -0,0 +1,177 @@
+// Implementation of Breadth First Search for Graph DS (adjacency list)
+
+#include<stdio.h>
+#include<stdlib.h>
+#include<time.h>
+
+typedef struct node{
+	int dest;
+	int weight;
+	struct node* next;
+}node; // a struture to denote vertices in a edgelist
+
+typedef struct adjList{
+	int startIndex;
+	int visited;

```

Comment:
Overall logic of the code is correct but correct choice of data type and data structure for implementation is suggested. It increases code readability and lowers time and space complexity :)